### PR TITLE
convert TTL > 30 days to unix epoch time for couchbase

### PR DESCRIPTION
**Pull request by woodpig07**
context see #46

### CODE REVIEW

**Comment by woodpig07** [comment]
@xavierchow To write unit test, is there any way to programmingly get couchbase doc meta? 
Something like
```
"meta":{
               "id":"test",
               "rev":"4-0000003f04e86b040000000000000000",
               "expiration":0,
               "flags":0
            },
```
`Model.findById` doesn't return it, I guess `loopback-connector-couchbase3` hides it?

Also I found out couchbase JDK has a limit on the value we can pass in `expiry`
https://github.com/couchbase/couchnode/blob/v2.6.3/lib/bucket.js#L1622
```
if (options.expiry < 0 || options.expiry > 2147483647) {
      throw new TypeError('expiry option needs to between 0 and 2147483647.');
    }
```
`2147483647` is `Tue Jan 19 2038 11:14:07 GMT+0800`, I guess we could ignore it for now?

**Comment by woodpig07** [comment]
Also could we drop support for node v4.x.x and v5.x.x?
The CI failed on node 4 and 5, but passed on node 6 because npm install `loopback-connector-couchbase3@0.3.1` which list `"couchbase": "^2.2.2"` in dependency, but it actually gets `couchbase@2.6.3`  installed, this lib has newer js syntax that's not working in node v4.x.x and v5.x.x.

**Comment by xavierchow** [comment]
> To write unit test, is there any way to programmingly get couchbase doc meta

Noop as I know, unless you create a view for the meta, but it's too much. instead of checking the meta ,can we mock/stub on the connector library function to check if the right value is passed in?


 > `2147483647` is `Tue Jan 19 2038 11:14:07 GMT+0800`, I guess we could ignore it for now

Sure, just ignore it .

**Comment by xavierchow** [comment]
> Also could we drop support for node v4.x.x and v5.x.x?

Yes, but you need a major upgrade about the version.also update the readme to say we don't support node v4/v5 anymore.

File: README.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ To use the `redis` backend, setup a model using the [redis connector](https://gi
 }
 ```
 
-To use the `couchbase` backend, setup a model using the [couchbase3 connector](https://github.com/Wiredcraft/loopback-connector-couchbase3) and use the mixin `CacheModel` with the model. The TTL is in seconds usually but also has a special case, see http://docs.couchbase.com/sdk-api/couchbase-node-client-2.1.2/Bucket.html#touch.
+To use the `couchbase` backend, setup a model using the [couchbase3 connector](https://github.com/Wiredcraft/loopback-connector-couchbase3) and use the mixin `CacheModel` with the model. The TTL is in seconds, for value larger than 302460*60 seconds (30 days), it would be converted to absolute times (from the epoch) in seconds, see http://docs.couchbase.com/sdk-api/couchbase-node-client-2.1.2/Bucket.html#touch.
```
Comment (xavierchow) [inline README.md]: `302460*60 ` => this is a format issue in CB document, actually it's 30 * 24 * 60 * 60.

File: backends/couchbase.js [hunk 2]
```diff
@@ -1,5 +1,12 @@
 'use strict';
 
+const ttlFixer = function(ttl) {
+  if (ttl > 30 * 24 * 60 * 60) {
+    return Date.now() / 1000 + ttl;
```
Comment (xavierchow) [inline backends/couchbase.js]:
```suggestion
    return Math.floor(Date.now() / 1000) + ttl;
```

File: backends/couchbase.js [hunk 3]
```diff
@@ -11,7 +18,7 @@ module.exports = function(Model, options) {
       // @see https://github.com/Wiredcraft/loopback-connector-couchbase3
       if (ctx.Model.getConnector().name === 'couchbase3') {
         // @see http://docs.couchbase.com/sdk-api/couchbase-node-client-2.1.2/Bucket.html#touch
-        ctx.options.expiry = options.ttl;
+        ctx.options.expiry = ttlFixer(options.ttl);
```
Comment (xavierchow) [inline backends/couchbase.js]:
```suggestion
        if (options.ttl > 30 * 24 * 60 * 60) {
          ctx.options.expiry = convertToAbsExpiry(options.ttl);
        } else {
          ctx.options.expiry = options.ttl;
         }
```

**Comment by woodpig07** [comment]
> > To write unit test, is there any way to programmingly get couchbase doc meta
> 
> Noop as I know, unless you create a view for the meta, but it's too much. instead of checking the meta ,can we mock/stub on the connector library function to check if the right value is passed in?

I can't figure out a easy way to mock connector library, so I ended up create view to get meta for testing.

**Comment by woodpig07** [comment]
> > Also could we drop support for node v4.x.x and v5.x.x?
> 
> Yes, but you need a major upgrade about the version.also update the readme to say we don't support node v4/v5 anymore.

I removed node 4 and 5 from CI process.
This library alone can support v4 and v5, but its peer dependency 
`loopback-connector-couchbase3@0.3.1` fails on v4 and v5.

**Comment by xavierchow** [comment]
Thank you @woodpig07

**Pull request merged**